modules/caddyhttp/matchers.go

```diff
@@ -559,7 +559,7 @@ func (m *MatchNot) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	for d.Next() {
 		var mp matcherPair
 		matcherMap := make(map[string]RequestMatcher)
-		for d.NextBlock(0) {
+		for d.NextArg() || d.NextBlock(0) {
```

Comment:
This is clever, but I do admit I think it's kinda weird. We can go with it for now and fix it later if people complain.

Reply:
What about it do you find weird?

Reply:
It's just an unusual condition for a parsing loop.

Reply:
Idk, reads perfectly fine to me. 🤷‍♂

"While there's still an arg or a block ahead, loop to read for additional matchers"